php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -2315,6 +2320,22 @@ abstract class NDB_BVL_Instrument extends NDB_Page
         );
     }
 
+    /**
+     * Parses the metadata of a LINST instrument and sets the proper variables
+     * in the instrument class accordingly
+     *
+     * This is intended to be overridden by a subclass of NDB_BVL_Instrument
+     * to add support for other file formats.
+     *
+     * @param string $filename The filename to be loaded, or a base64 encoded
+     *                         string.
+     *
+     * @return void
+     */
+    function loadInstrumentMetadata(string $filename): void
```

Comment:
is this maybe supposed to be an abstract function?

Reply:
I made it to be empty just to be consistent with the function below `loadInstrumentFile()`

In practice however, making this abstract would require for every project to implement this function for every instrument which extends this class... it would be too costly for projects

Reply:
truee